configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java

```diff
@@ -557,24 +561,42 @@ private void cleanupTempDirectory(File tempDir) {
         }
     }
 
-    void redeployAllApplications() throws InterruptedException {
+    boolean redeployAllApplications(Duration maxDuration) throws InterruptedException {
+        Instant end = Instant.now().plus(maxDuration);
+        Set<ApplicationId> applicationIds = listApplications();
```

Comment:
Would be easier to read if you called this "applicationsNotRedeployed".

Reply:
Agreed, will fix later